src/XAD/Jacobian.hpp

```diff
@@ -0,0 +1,79 @@
+/*******************************************************************************
+
+   Implementation of Jacobian computing methods.
+
+   This file is part of XAD, a comprehensive C++ library for
+   automatic differentiation.
+
+   Copyright (C) 2010-2024 Xcelerit Computing Ltd.
+
+   This program is free software: you can redistribute it and/or modify
+   it under the terms of the GNU Affero General Public License as published
+   by the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU Affero General Public License for more details.
+
+   You should have received a copy of the GNU Affero General Public License
+   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+******************************************************************************/
+
+#include <functional>
+#include <vector>
+
+namespace xad
+{
+
+template <class T>
+class Jacobian
+{
+  public:
+    Jacobian(std::vector<std::function<T(std::vector<T> &)>> foos, std::vector<T> &v,
```

Comment:
Could you explain why you need multiple functions here? Shouldn't that be a single function with multiple outputs instead, i.e. 

```c++
std::function<std::vector<T>(const std::vector<T>&)>
```

Reply:
I was skipping a step for the sake of testing. I'll change that once I get the Jacobian working.